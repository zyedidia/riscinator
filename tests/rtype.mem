fff04093
fff04113
fff04193
fff04213
fff04293
fff04313
fff04393
fff04413
fff04493
fff04513
fff04593
fff04613
fff04693
fff04713
fff04793
fff04813
fff04893
fff04913
fff04993
fff04a13
fff04a93
fff04b13
fff04b93
fff04c13
fff04c93
fff04d13
fff04d93
fff04e13
fff04e93
fff04f13
fff04f93
01ff00b3
41ff0133
401101b3
41f10233
004162b3
0012e333
01f173b3
0062f433
0012c4b3
01efc533
0062c5b3
00229613
00261693
00568733
00671793
405fd813
4015d893
4027d913
0047d993
00afda13
01ffda93
01478d33
40fa0fb3
01afaeb3
01fd2f33
