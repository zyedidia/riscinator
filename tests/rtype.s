not x1, x0
not x2, x0
not x3, x0
not x4, x0
not x5, x0
not x6, x0
not x7, x0
not x8, x0
not x9, x0
not x10, x0
not x11, x0
not x12, x0
not x13, x0
not x14, x0
not x15, x0
not x16, x0
not x17, x0
not x18, x0
not x19, x0
not x20, x0
not x21, x0
not x22, x0
not x23, x0
not x24, x0
not x25, x0
not x26, x0
not x27, x0
not x28, x0
not x29, x0
not x30, x0
not x31, x0
add x1, x30, x31
sub x2, x30, x31
sub x3, x2, x1
sub x4, x2, x31
or x5, x2, x4
or x6, x5, x1
and x7, x2, x31
and x8, x5, x6
xor x9, x5, x1
xor x10, x31, x30
xor x11, x5, x6
slli x12, x5, 2
slli x13, x12, 2
add x14, x13, x5
slli x15, x14, 6
srai x16, x31, 5
srai x17, x11, 1
srai x18, x15, 2
srli x19, x15, 4
srli x20, x31, 10
srli x21, x31, 31
add x26, x15, x20
sub x31, x20, x15
slt x29, x31, x26
slt x30, x26, x31

