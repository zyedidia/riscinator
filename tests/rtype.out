0: 0
1: -2
2: 0 
3: 2
4: 1
5: 1
6: -1
7: 0
8: 1
9: -1
10: 0
11: -2
12: 4
13: 16
14: 17
15: 1088
16: -1
17: -1
18: 272
19: 68
20: 4194303
21: 1
22: -1
23: -1
24: -1
25: -1
26: 4195391 
27: -1
28: -1
29: 1
30: 0
31: 4193215
